### PR TITLE
iprov/CHAT-1018 : add remove option to upload files on the chat room

### PR DESCRIPTION
**Pull request by waelghannem**
i added new condition to displayActions and messageActions functions to add delete option when type = 'type-file'

### CODE REVIEW

File: application/src/main/webapp/vue-app/components/ExoChatMessageDetail.vue [hunk 1]
```diff
@@ -209,13 +209,17 @@ export default {
       return eXo.chat.userSettings.username === this.message.user;
     },
     messageActions() {
+      if(this.message.options && this.message.options.type === 'type-file') {
+        return messageActions.filter( messageAction => messageAction.key !== 'edit');
+      }
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/vue-app/components/ExoChatMessageDetail.vue]: This filter must be done in the action definition directly, in application/src/main/webapp/vue-app/extension.js, in the `enabled` property.

File: application/src/main/webapp/vue-app/components/ExoChatMessageDetail.vue [hunk 2]
```diff
@@ -209,13 +209,17 @@ export default {
...
       return messageActions.filter(menu => !menu.enabled || menu.enabled(this));
     },
     displayUserInformation() {
       const messageType = this.message.options ? this.message.options.type : null;
       return messageType !== chatConstants.ROOM_MEMBER_LEFT && messageType !== chatConstants.REMOVE_TEAM_MESSAGE && messageType !== chatConstants.ADD_TEAM_MESSAGE && !this.hideAvatar && !this.isCurrentUser;
     },
     displayActions() {
+      if(this.message.options && this.message.options.type === 'type-file'){ return true;}
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/vue-app/components/ExoChatMessageDetail.vue]: Is this change really necessary ? The method displayActions allows to define if the whole menu actions on the message must be displayed or not. In the case of a type `type-file`, the condition `!this.miniChat && this.message.type !== chatConstants.DELETED_MESSAGE && this.messageActions && this.messageActions.length` should already return true.

Comment (waelghannem) [reply application/src/main/webapp/vue-app/components/ExoChatMessageDetail.vue]: no, in the case of a type 'type-file', the condition !this.miniChat && this.message.type !== chatConstants.DELETED_MESSAGE && this.messageActions && this.messageActions.length returns false.

Comment (thomasdelhomenie) [reply application/src/main/webapp/vue-app/components/ExoChatMessageDetail.vue]: Yes, but it should return true. `messageActions` should not be empty, it should contain the delete action at least. We should not introduce a specific test for the particular type of type-file IMO.

File: application/src/main/webapp/vue-app/extension.js [hunk 3]
```diff
@@ -524,7 +524,9 @@ const DEFAULT_MESSAGE_ACTIONS = [
         !comp.message.isSystem &&
         !comp.message.isDeleted &&
         !comp.message.notSent &&
-        comp.isCurrentUser
+        comp.isCurrentUser ||
+        comp.message.options &&
+        comp.message.options.type === 'type-file'
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/vue-app/extension.js]: As discussed, we should avoid testing on specific type if possible. I guess the delete action does not display because the file type message are considered as system, right ? If so, it should not IMO, so we must make file type message as non-system message.

**Comment by mzorai** [comment]
>     * ve notes
For attachments, only delete action is needed

**Comment by thomasdelhomenie** [comment]
@waelghannem works fine :+1: 
@mzorai WDYT about:

> Also, it works only for new messages, not for existing messages. Should we do a data upgrade to make it work for existing messages ?

**Comment by mzorai** [comment]
> @waelghannem works fine +1
> @mzorai WDYT about:
> 
> > Also, it works only for new messages, not for existing messages. Should we do a data upgrade to make it work for existing messages ?

Yes we should make the delete action for uploaded files available even for old messages

**Comment by mzorai** [comment]
> > @waelghannem works fine +1
> > @mzorai WDYT about:
> > > Also, it works only for new messages, not for existing messages. Should we do a data upgrade to make it work for existing messages ?
> 
> Yes we should make the delete action for uploaded files available even for old messages

After discussing with @thomasdelhomenie  and @shazia, we concluded that no need to make the delete action available for "old" message. A user will not intend to come back to old messages and delete old attachement
